packages/frontend/src/hooks/cartesianChartConfig/useCartesianChartConfig.ts

```diff
@@ -254,6 +309,15 @@ const useCartesianChartConfig = ({
     useEffect(() => {
         if (availableFields.length > 0) {
             setDirtyLayout((prev) => {
+                const sortedDimensions =
+                    availableDimensions.length > 1
+                        ? sortDimensions(
+                              availableDimensions,
+                              explore,
+                              columnOrder,
+                          )
+                        : availableDimensions;
```

Comment:
What if we did this in the hook that returns the `availableDimensions` so it's  already sorted ?

Reply:
I would have moved the `availableDimensions.length > 1` condition inside the sort function.